File: Data Structures & Algorithms/insert-new-interval/submission-5.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.sort()
        ret = []
        for x,y in intervals:
            if(x <= newInterval[0] <= y or newInterval[0] <= x <= newInterval[1]):
                newInterval[0] = min(newInterval[0], x)
                newInterval[1] = max(newInterval[1], y)
            elif(y < newInterval[0]):
                ret.append([x,y])
            else:
                ret.append(newInterval)
                newInterval = [x,y]
        ret.append(newInterval)
        return ret

File: Data Structures & Algorithms/merge-intervals/submission-5.js
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number[][]}
     */
    merge(intervals) {
        intervals.sort((a,b) => a[0] - b[0])
        let ret = [intervals[0]]
        for(let [x,y] of intervals){
            if(x <= ret[ret.length - 1][1]){
                let minVal = Math.min(x, ret[ret.length - 1][0])
                let maxVal = Math.max(y, ret[ret.length - 1][1])
                ret[ret.length - 1][0] = minVal
                ret[ret.length - 1][1] = maxVal
            }else{
                ret.push([x,y])
            }
        }
        return ret
    }
}
